pkg/watch/errorlist.go

```diff
@@ -15,22 +15,65 @@ limitations under the License.
 
 package watch
 
-import "strings"
+import (
+	"fmt"
+	"strings"
 
-// errorList is an error that aggregates multiple errors.
-type errorList []error
+	"k8s.io/apimachinery/pkg/runtime/schema"
+)
+
+// errorList is an error that aggregates multiple gvkErrs.
+type errorList struct {
+	errs        []gvkErr
+	isUniversal bool
```

Comment:
"universal" seems to be derived from Max's previous comment, though out of the context of the PR review I think its meaning is unclear.

Is there a way we could rename this to convey the difference between the two cases?

In particular, it might be good for the `errorList` to have its default setting be the "universal" case and have the question of "is a failing GVKs error" only be true if GVK-specific errors have been specified in WatchManager.

This would mean that a future engineer who does the following:

```
var errList errorList

// Do something that can error in a non-GVK specific way
err := prerequisiteLogic()
if err != nil {
  errList.Add(err)
  return errList
}
```

Would return an error list that is a "general" or "universal" error, without them having to explicitly say that it is.  Because they didn't specify a GVK, it is inherently a "universal" error.

Does that sound like the functionality we are going for?

Reply:
Not quite IIRC @maxsmythe correct me if I stray from the right path.

There can be errors associated with a GVK, down the road, that are "universal" or "global" because some other thing or logic downstream to WatchManager is failing. So the mutual exclusion wouldn't make sense in this case.

Reply:
Per our chat offline, I added some convenience methods and moved on to make it more obvious/ easy to use `errorList`. Lmk what you think @julianKatz . I also want to make sure we don't over-engineer this piece of the codebase 😛